contentcuration/contentcuration/frontend/shared/data/resources.js

```diff
@@ -852,6 +885,47 @@ export const Session = new IndexedDBResource({
       }
     },
   },
+  channelSyncKeepAlive(channelId) {
+    document.hasFocus()
+      ? this.updateSession({ [`${ACTIVE_CHANNELS}.${channelId}`]: Date.now() })
+      : null;
+  },
+  async setChannelScope() {
+    const channelId = (window.CHANNEL_EDIT_GLOBAL || {}).channel_id || null;
+    if (channelId) {
+      channelScope.id = channelId;
+      const channelRev = (window.CHANNEL_EDIT_GLOBAL || {}).channel_rev || 0;
+      await this.updateSession({
+        [`${MAX_REV_KEY}.${channelId}`]: channelRev,
+      });
+      this.channelSyncKeepAlive(channelId);
+      setInterval(() => this.channelSyncKeepAlive(channelId), CHANNEL_SYNC_KEEP_ALIVE_INTERVAL);
```

Comment:
Yeah, we wouldn't need to continue doing these checks if we're not online for example - but it might be more complexity than it's worth to monitor it.

Reply:
Have made some updates to at least sleep on blur.